### ISSUE TITLE
performance issue in binders with asterisk in the name

### ISSUE THREAD

**Issue by stalniy** [mikeric/rivets#595]
binders with asterisk in the name slows down the execution of binders almost in 2 times!

Proff: http://jsperf.com/rivets-each-and-star-binding/6

@blikblum @Leeds-eBooks @Duder-onomy

**Comment by jccazeaux** [comment]
Your cases are not comparable. On one side you have
```html
<li rv-each-item="items" rv-index="index">{ item }</li>
 ```
On the other side you have
```html
<li rv-each-item="items">{ item }</li>
 ```

The performance difference is because you have one binder in first case, two binders in second case.
I tried the test with two binders in both cases : http://jsperf.com/rivets-each-and-star-binding/8
One is 
```html
<li rv-each-item="items" rv-index="index">{ item }</li>
 ```
and the other
```html
<li rv-each-item="items" rv-attr-index="index">{ item }</li>
 ```
Where `rv-attr-index` is a custom binder without `*`.
Performances are very similar.

**Comment by stalniy** [comment]
You are right at some point. But your test is also not comparable. Because you use binder with asterisk in both cases. 

So, create a binder with name `index` instead which sets `index` attribute. 

The root of the performance issue is in asterisk. Rivets can't much binder with asterisk efficiently so it iterates over all binders in order to find appropriate. More binders you have in `rivets.bibders` bigger performance impact because of iteration over object with bigger amount of properties

I'll create one more test case and post it here

**Comment by jccazeaux** [comment]
A solution to improve performance would be to put `*` binders in a separate array. Is it really worth it?

**Comment by stalniy** [comment]
The better solution could be this:
1. Check if binder can be found by attr name
2. If not, try simple matching algorithm: get the first word of attr (attr name is words separated by minus) and append -* at the end. Check if this binder exists 
3. If not, iterate over all binders and match appropriate

This algorithm will cover almost all binders with asterisk in name. 

I'll update jsperf a bit later with proposed solution
